esp-wifi/src/compat/misc.rs

```diff
@@ -158,11 +158,37 @@ unsafe extern "C" fn strdup(str: *const i8) -> *const u8 {
 unsafe extern "C" fn atoi(str: *const i8) -> i32 {
     trace!("atoi {:?}", str);
 
-    unsafe {
-        let s = core::ffi::CStr::from_ptr(str);
-        let r = s.to_str().unwrap().parse().unwrap();
-        r
+    let mut sign: i32 = 1;
+    let mut res: i32 = 0;
+    let mut idx = 0;
+
+    while str.add(idx).read_volatile() as u8 == b' ' {
```

Comment:
Please add a comment about what each block does, in case someone misses that this is an `==` and not a `!=` :D